akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -441,3 +445,102 @@ private[akka] final class ReducerState[T, R](val collector: java.util.stream.Col
   def finish(): R = collector.finisher().apply(reduced)
 }
 
+/**
+ * INTERNAL API
+ */
+final private[stream] class LazySink[T, M](sinkFactory: T ⇒ Future[Sink[T, M]], zeroMat: () ⇒ M) extends GraphStageWithMaterializedValue[SinkShape[T], Future[M]] {
```

Comment:
Well, but that can't solve the Future[Source] case though. So that would mean we would have two versions for the same thing:

```scala
src.lazy
Source.lazy(x => srcFuture)
```

So I opt for the wrapping API since it is more general, and the use-case for this is rare enough to not warrant a sugared `.lazy` method.